tests/spec/test_blueprint.py

```diff
@@ -0,0 +1,304 @@
+from enum import Enum
+import re
+from typing import Optional, List, Any, Mapping
+
+import pytest
+from flask import Flask, Blueprint
+from openapi_spec_validator import OpenAPIV31SpecValidator
+from pydantic import BaseModel, StrictFloat, Field, RootModel
+
+from flask_pydantic_spec import Response
+from flask_pydantic_spec.types import FileResponse, Request, MultipartFormRequest
+from flask_pydantic_spec import FlaskPydanticSpec
+from flask_pydantic_spec.config import Config
+from flask_pydantic_spec.utils import get_model_name
+
+from tests.common import ExampleConverter, UnknownConverter
+
+
+class ExampleModel(BaseModel):
+    name: str = Field(json_schema_extra={"strip_whitespace": True})
+    age: int
+    height: StrictFloat
+
+
+class TypeEnum(str, Enum):
+    foo = "foo"
+    bar = "bar"
+
+
+class ExampleQuery(BaseModel):
+    query: str
+    type: Optional[TypeEnum] = None
+
+
+ExampleNestedList = RootModel[List[ExampleModel]]
+
+
+class ExampleNestedModel(BaseModel):
+    example: ExampleModel
+
+
+class ExampleDeepNestedModel(BaseModel):
+    data: List["ExampleModel"]
+
+
+@pytest.fixture
+def empty_app():
+    return Flask(__name__)
+
+
+@pytest.fixture
+def app_name():
+    return "flask"
+
+
+@pytest.fixture
+def blueprint_name():
+    return "blueprint"
+
+
+@pytest.fixture
+def empty_blueprint(blueprint_name):
+    return Blueprint(blueprint_name, __name__, url_prefix="/blueprint")
+
+
+@pytest.fixture
+def api() -> FlaskPydanticSpec:
+    return FlaskPydanticSpec(
+        "flask",
+        tags=[{"name": "lone", "description": "a lone api"}],
+        validation_error_code=400,
+    )
+
+
+@pytest.fixture
+def bp_api(api: FlaskPydanticSpec, empty_blueprint: Blueprint) -> FlaskPydanticSpec:
+    return api.for_blueprint(
+        empty_blueprint,
+        tags=[{"name": "lone", "description": "a lone api"}],
+    )
+
+
+def test_spectree_init():
+    spec = FlaskPydanticSpec(path="docs")
+    conf = Config()
+
+    assert spec.config.TITLE == conf.TITLE
+    assert spec.config.PATH == "docs"
+
+
+def test_register(app_name, empty_app, empty_blueprint):
+    api = FlaskPydanticSpec(app_name)
+    api.register(empty_app)
+    blueprint_api = FlaskPydanticSpec("blueprint_api")
+
+    blueprint_api.register(empty_blueprint)
+    empty_app.register_blueprint(empty_blueprint)
+
+
+def test_spec_generate(app_name, empty_app, empty_blueprint):
+    api = FlaskPydanticSpec(app_name)
+    api.register(empty_app)
+
+    bp_spec = api.for_blueprint(
+        empty_blueprint,
+        title=empty_blueprint.name,
+        info={"title": "override", "description": "blueprint level description"},
+    )
+    empty_app.register_blueprint(empty_blueprint)
```

Comment:
I wonder if it's worth adding a `# routes would be defined here` comment as I found it took me a minute to figure that out from the larger test as there was more to scan.

Reply:
Would it be better to pull the fixtures that are all about setting up an app/blueprint out of these files?